src/main/java/core/org/akaza/openclinica/dao/core/CoreResources.java

```diff
@@ -405,14 +405,17 @@ public static String getRequestSchema(HttpServletRequest request) {
     }
 
     public static boolean setRequestSchema(String schema) {
-        ServletRequestAttributes requestAttributes = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
-        if (requestAttributes != null && requestAttributes.getRequest() != null) {
-            HttpServletRequest request = requestAttributes.getRequest();
-            request.setAttribute("requestSchema", schema);
-            return true;
-        } else {
-            CoreResources.tenantSchema.set(schema);
-        }
+    	if(schema!=null && schema.trim().length() > 0) {
```

Comment:
instead of adding a null check here, make sure that the calling method is passing a non-null schema. This is masking the root cause of the problem.

Reply:
done